### DISCUSSION TITLE
ModuleFederationPlugin can not directly get shared module default export in .mjs file

### DISCUSSION THREAD

**Discussion by lzxb** [webpack/webpack#17906] [Q&A]
# Bug report

**What is the current behavior?**
Failed to obtain the correct default export
**If the current behavior is a bug, please provide the steps to reproduce.**

<img width="436" alt="image" src="https://github.com/webpack/webpack/assets/8424643/01b3c0ae-4cf5-4612-9bad-7d8a6ec8ef72">
<img width="571" alt="image" src="https://github.com/webpack/webpack/assets/8424643/67997931-58a9-49fd-a50c-2fd824497b4b">

After the build was completed, the default value was lost

![image](https://github.com/webpack/webpack/assets/8424643/7bef0bb5-384a-41fc-9ab0-b75972326ee3)

![image](https://github.com/webpack/webpack/assets/8424643/a672ddce-4fcf-4015-b7f9-461a5e697a72)

**What is the expected behavior?**
Get exported objects

**Other relevant information:**
webpack version:  5.88.2
Node.js version:v18.16.1
Operating System: mac os
Additional tools:

**Comment by alexander-akait** [comment]
Please create reproducible test repo I will show solutions, thank you

**Reply by barabaiiika** [reply]
@alexander-akait I created a simple repository to reproduce the incorrect behavior https://github.com/barabaiiika/missing_default_export_example
I will be happy if you find time for this☺️

**Reply by alexander-akait** [reply]
I think it is duplicate of https://github.com/webpack/webpack/issues/16125#event-15952149896, sorry for delay
